File: backend/app/infrastructure/repositories/mappers/user_mapper.py
from app.domain.entities.user import User as UserDomain
from app.infrastructure.repositories.orm.user import User as UserORM

def to_domain(orm_user: UserORM) -> UserDomain:
    return UserDomain(
        id=orm_user.id,
        username=orm_user.username,
        email=orm_user.email,
        hashed_password=orm_user.hashed_password,
        profile_picture_url=orm_user.profile_picture_url,
        created_at=orm_user.created_at
    )

File: frontend/lib/data/mappers/user_mapper.dart
// frontend/lib/data/mappers/user_mapper.dart
import 'package:frontend/data/models/user_model.dart';
import 'package:frontend/domain/entities/user.dart';

class UserMapper {
  static User fromModel(UserModel model) {
    return User(
      id: model.id,
      username: model.username,
      email: model.email,
      profilePictureUrl: model.profilePictureUrl,
      createdAt: model.createdAt,
    );
  }

  static UserModel toModel(User entity) {
    return UserModel(
      id: entity.id,
      username: entity.username,
      email: entity.email,
      profilePictureUrl: entity.profilePictureUrl,
      createdAt: entity.createdAt,
    );
  }
}
